Reneiski_Homework1/ReneiskiSelenium11/Program.cs

```diff
@@ -0,0 +1,246 @@
+            var editedUserSalaryTextBox = _driver.FindElement(By.XPath("//*[@id='salary']"));
+            editedUserSalaryTextBox.Clear();
+            editedUserSalaryTextBox.SendKeys(editedUserSalary);
+
+            var submitButtonAfterEdit = _driver.FindElement(By.XPath("//*[@id='submit']"));
+            submitButtonAfterEdit.Click();
+
+            var deleteButton = _driver.FindElement(By.XPath("//*[@id='delete-record-4']"));
+            deleteButton.Click();
+
+            var currentResult2 = _driver.FindElement(By.XPath("//*[@class='rt-table']"));
+            var currentResultText2 = currentResult2.Text;
+
+            Assert.IsFalse(currentResultText2.Contains(firstName));
+            Assert.IsFalse(currentResultText2.Contains(lastName));
+            Assert.IsFalse(currentResultText2.Contains(userAge));
+            Assert.IsFalse(currentResultText2.Contains(userEmail));
+            Assert.IsFalse(currentResultText2.Contains(userSalary));
+            Assert.IsFalse(currentResultText2.Contains(userDepartment));
+        }
+
+        [Test]
+        public void ForthButtonsTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/buttons");
+
+            var doubleClickButton = _driver.FindElement(By.XPath("//*[@id='doubleClickBtn']"));
+            _driverActions.DoubleClick(doubleClickButton).Perform();
+
+            var doubleClickResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='doubleClickMessage']")));
+            var doubleClickResultText = doubleClickResult.Text;
+
+            Assert.IsTrue(doubleClickResultText.Contains("You have done a double click"));
+
+            var rightClickButton = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='rightClickBtn']")));
+            _driverActions.ContextClick(rightClickButton).Perform();
+
+            var rightClickResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='rightClickMessage']")));
+            var rightClickResultText = rightClickResult.Text;
+
+            Assert.IsTrue(rightClickResultText.Contains("You have done a right click"));
+
+            var dynamicClickButton = _driverWait.Until(drv => drv.FindElement(By.XPath("/html/body/div[2]/div/div/div[2]/div[2]/div[2]/div[3]/button")));
+            dynamicClickButton.Click();
+
+            var dynamicClickResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='dynamicClickMessage']")));
+            var dynamicClickResultText = dynamicClickResult.Text;
+
+            Assert.IsTrue(dynamicClickResultText.Contains("You have done a dynamic click"));
+        }
+
+        [Test]
+        public void FifthLinksTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/links");
+
+            var simpleLink = _driver.FindElement(By.XPath("//*[@id='simpleLink']"));
+            simpleLink.Click();
+
+            var currentUrl = _driver.FindElement(By.XPath("//p/a[@href='https://demoqa.com']"));
```

Comment:
Actually, this check doesn't make a lot of sense because you find element on the first opened tab by href and then gets value of href attibute.

You would better to move demoqa.com url into const on a class or test level and use in the expected result when compare url of new opened tab vs expected url

Reply:
Fixed